api/types/accesslist/review.go

```diff
@@ -0,0 +1,174 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package accesslist
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/api/types/header"
+	"github.com/gravitational/teleport/api/types/header/convert/legacy"
+)
+
+// Review is an access list review resource.
+type Review struct {
```

Comment:
So this is implemented the same way as AccessList which caused issues for terraform provider. Maybe I'm not up-to-date on how we're doing resources now, but the way we used to do it is take a protobuf-generated resource and make sure it satisfies types.Resource interface and use it internally. E.g. look at all the constructors in api/types, like `Database`.

This struct doesn't implement `types.Resource` and has to be converted back and forth from the proto struct. What's the benefit?

Reply:
Talking over Slack, this is basically an unfortunate side effect of not having gogo's custom parsing/marshaling/casting/etc. options any more.